src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -242,30 +258,30 @@ public static ProjectStateChecksums Deserialize(ObjectReader reader)
     }
 }
 
-internal sealed class DocumentStateChecksums(Checksum infoChecksum, Checksum textChecksum) : IChecksummedObject
+internal sealed class DocumentStateChecksums(
+    DocumentId documentId,
+    Checksum infoChecksum,
+    Checksum textChecksum) : IChecksummedObject
 {
     public Checksum Checksum { get; } = Checksum.Create(infoChecksum, textChecksum);
+
+    public DocumentId DocumentId => documentId;
     public Checksum Info => infoChecksum;
     public Checksum Text => textChecksum;
 
-    public void AddAllTo(HashSet<Checksum> checksums)
-    {
-        checksums.AddIfNotNullChecksum(this.Checksum);
-        checksums.AddIfNotNullChecksum(this.Info);
-        checksums.AddIfNotNullChecksum(this.Text);
-    }
```

Comment:
only used in tests.  moved there.  will do the same with ProjectStateChecksums in a later PR.